src/ripple/app/tx/Transaction.h

```diff
@@ -124,6 +124,33 @@ class Transaction
         mInLedger = ledger;
     }
 
+    /**
+     * Set this flag once an attempt has been made to apply this transaction
+     * within a batch.
+     */
+    void setBatched()
+    {
+        mBatched = true;
+    }
+
+    /**
+     * Detect if transaction has been batched, but clear the flag in case
+     * another attempt is made to process in a batch.
+     *
+     * @return whether transaction has been batched.
+     */
+    bool getClearBatched()
+    {
+        if (mBatched)
```

Comment:
Some weird braces there.  ;-)  Was there supposed to be a lock?

Consider:

    if (! mBatched)
        return false;
   mBatched = false;
   return true;